usersync/cookie_test.go

```diff
@@ -130,7 +125,7 @@ func TestParseCorruptedCookieJSON(t *testing.T) {
 }
 
 func TestParseNilSyncMap(t *testing.T) {
```

Comment:
I don't understand what the intention is of this test. What is a nil sync map with respect to a cookie? Is it the fact that this test does not define `uids` in the cookie? It seems like this test was previously causing a JSON unmarshal error in the function under test because the birthday type was an invalid format. Now we aren't getting a JSON unmarshal error because this invalid field `optoutt` will be ignored.  You can force a JSON unmarshal error by instead doing `"{\"optout\":123}` but the JSON unmarshal error code path is already covered by `TestParseCorruptedCookieJSON` which seems redundant. If the concern here is to just verify consistency using `ensureConsistency` when `uids` is missing in the cookie, then what you have here should suffice even though a different code path is taken now.

Reply:
I'm also unsure what the purpose of this test is, I just updated it since it started failing when birthday was removed. I don't know why the test chose to trigger an unmarshal error so that an empty cookie can be returned through NewCookie(). Why not just initialize an empty cookie in the beginning?

Reply:
Agreed, if we are going to keep this test, I would just initialize to an empty cookie by setting `Value` to an empty string. Right now this test is a duplicate of `TestParseCorruptedCookieJSON` so I suggest making that change.